app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -44,13 +44,19 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 
     curl --insecure --output $CONSUMER_RPM <%= subscription_manager_configuration_url %>
     yum localinstall $CONSUMER_RPM -y
-    yum install subscription-manager -y
     subscription-manager register --org=<%= @organization.label %> --activationkey="sample-key"
-
-    rm -f $CONSUMER_RPM
```

Comment:
Why do you not want to clean this up?

Reply:
When host is created, I want to check if everything went well or exit the script. Because we use two methods for host registration (`create_host_curl` &  `subscription-manager`), the easiest way was to check status of the last command. See:
```
if ! [ $? = 0 ]; then
    echo ""
    echo "Host registration failed"
    exit 1
fi
```
It worked well for `create_host_curl`, but not for the case with katello & rhel/centos, where last command is `rm -f $CONSUMER_RPM`.

And since the size of file is only `8.6K` and it's created in `/tmp` folder (where it will be deleted after a while, right?), I thought that keeping the file there won't be a problem. So now last command is `subscription-manager register`.

Plus we don't delete it also in the instructions _Hosts > Content Hosts > Register Content Hosts_

Reply:
I'd suggest to avoid making the whole script idempotent. That's hard and I wouldn't want to put that burden on users who want to modify this. See https://github.com/theforeman/foreman/pull/7976#discussion_r486968427 for my suggestion to avoid needing this.

Reply:
we could save the result of curl/sub-man call to a variable and check that instead of $?

Reply:
Looks like I didn't submit my comment, but the easier solution is to first `rm -f $CONSUMER` and then run `subscription-manager`. Then you still have `$?`. That's a good thing regardless, but storing is a good option as well.

Another common pattern for this is:

```sh
die() {
  echo "$*"
  exit 1
}

my_command || die "Failed to do x"
```